src/orion/service/client/workon.py

```diff
@@ -0,0 +1,178 @@
+"""Minimal API to run workon"""
+import logging
+import numbers
+from typing import Dict, List, Optional
+
+from orion.core.worker.trial import TrialCM
+from orion.core.worker.trial_pacemaker import TrialPacemaker
+from orion.service.client.base import (
+    BaseClientREST,
+    ExperiementIsNotSetup,
+    RemoteException,
+    RemoteExperiment,
+    RemoteTrial,
+)
+
+log = logging.getLogger(__name__)
+
+
+class WorkonClientREST(BaseClientREST):
+    """Minimal REST API, only implements the required method to run workon.
+
+    Its goal is limited to communicating to the remote algo,
+    For generic query you should implement the functionality inside the rest storage.
+
+    """
+
+    def __init__(self, endpoint, token) -> None:
+        super().__init__(endpoint, token)
+        self.experiment = None
+        self._pacemakers = {}
+
+    @property
+    def experiment_name(self) -> Optional[str]:
+        """returns the current experiment name"""
+        return self.experiment.name if self.experiment else None
+
+    @property
+    def experiment_id(self) -> Optional[str]:
+        """returns the current experiment id"""
+        return self.experiment.euid if self.experiment else None
+
+    def new_experiment(self, name, **config) -> RemoteException:
+        """Create a new experiment"""
+        payload = self._post("experiment", name=name, **config)
+
+        payload.pop("experiment_name", None)
+        payload["name"] = name
+        self.experiment = RemoteExperiment(**payload)
+        return self.experiment
+
+    def suggest(self, pool_size: int = 1, experiment_name=None) -> TrialCM:
```

Comment:
Why should it support passing the experiment name as an argument? Shouldn't the WorkonClientREST be used only together with an existing experiment? After all it's used to replace `some_existing_experiment.workon`.